src/nvim/api/buffer.c

```diff
@@ -984,12 +985,23 @@ void nvim_buf_set_name(Buffer buffer, String name, Error *err)
 
   try_start();
 
+  const bool is_curbuf = buf == curbuf;
+  const int save_acd = p_acd;
+  if (!is_curbuf) {
+    // Temporarily disable 'autochdir' when setting file name for another buffer.
+    p_acd = false;
+  }
+
   // Using aucmd_*: autocommands will be executed by rename_buffer
   aco_save_T aco;
   aucmd_prepbuf(&aco, buf);
```

Comment:
should the `p_acd` save/restore be done by `aucmd_prepbuf` ?

Reply:
I think that'll break other APIs.